torch/fx/experimental/proxy_tensor.py

```diff
@@ -202,28 +217,43 @@ def __init__(self, tracer):
 
     def __torch_dispatch__(self, func_overload, types, args=(), kwargs=None):
         func = func_overload.overloadpacket
+        if func_overload == torch.ops.prim.device.default:
+            return args[0].device
         if any(tuple(isinstance(arg, ProxyTensor) for arg in pytree.tree_flatten(args)[0])):
             return proxy_call(func_overload, args, kwargs)
         else:
             proxy_out = self.tracer.create_proxy('call_function', func, args, kwargs,
                                                  name=self.tracer.graph._target_to_str(func.__name__))
 
-            with no_dispatch():
-                real_out = func_overload(*args, **kwargs)
+            real_out = func_overload(*args, **kwargs)
 
             return wrap_output(real_out, proxy_out)
 
 
-def make_fx(f, decomposition_table=None, trace_factory_functions=True):
+def make_fx(f, decomposition_table=None, trace_factory_functions=True, use_fake=False):
```

Comment:
what's preventing us from switching over to Fake completely? or making the default true here?

Reply:
Well, there's still a couple of extra failures with FakeTensors. But perhaps we could.